sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeExtractors.scala

```diff
@@ -210,4 +203,45 @@ case class GpuArrayContains(left: Expression, right: Expression)
     lhs.getBase.listContainsColumn(rhs.getBase)
 
   override def prettyName: String = "array_contains"
-}
\ No newline at end of file
+}
+
+object GetArrayItemUtil {
+  def evalColumnar(array: GpuColumnVector, ordinal: Scalar, dataType: DataType,
+                   zeroIndexed: Boolean): ColumnVector = {
+    // for array index use case, index starts at 0
+    if (zeroIndexed) {
+      // Need to handle negative indexes...
+      if (ordinal.isValid && ordinal.getInt >= 0) {
+        array.getBase.extractListElement(ordinal.getInt)
+      } else {
+        withResource(Scalar.fromNull(
+          GpuColumnVector.getNonNestedRapidsType(dataType))) { nullScalar =>
+          ColumnVector.fromScalar(nullScalar, array.getRowCount.toInt)
+        }
+      }
+    } else {
+      // for element_at use case, index starts at 1
+      if (ordinal.isValid) {
+        if (ordinal.getInt > 0) {
+          // SQL 1-based index
+          array.getBase.extractListElement(ordinal.getInt - 1)
+        } else if (ordinal.getInt == 0) {
+          throw new ArrayIndexOutOfBoundsException("SQL array indices start at 1")
+        } else {
+          array.getBase.extractListElement(ordinal.getInt)
+        }
+      } else {
+        withResource(Scalar.fromNull(
+          GpuColumnVector.getNonNestedRapidsType(dataType))) { nullScalar =>
+          ColumnVector.fromScalar(nullScalar, array.getRowCount.toInt)
+        }
+      }
+    }
+  }
+}
+
+object GetMapValueUtil {
+  def evalColumnar(map: GpuColumnVector, key: Scalar): ColumnVector = {
```

Comment:
Again public API needs doc

Reply:
I am wondering whether we really need this API since it is quite simple .

Reply:
Yes, currently map only supports StringType as key&value which looks simple. 
We may need more code to deal with other more types in the future.

Reply:
NIT. A standard doc would be better, including params, returns.